aspnetcore/fundamentals/http-requests.md

```diff
@@ -0,0 +1,367 @@
+
+To create a handler, a class can be added, deriving from `DelegatingHandler`. The `SendAsync` method can then be overridden to execute code before passing the request to the next handler in the pipeline:
+
+```csharp
+private class RetryHandler : DelegatingHandler
+{
+    public int RetryCount { get; set; } = 5;
+
+    protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+    {
+        for (var i = 0; i < RetryCount; i++)
+        {
+            try
+            {
+                return base.SendAsync(request, cancellationToken);
+            }
+            catch (HttpRequestException) when (i == RetryCount - 1)
+            {
+                throw;
+            }
+            catch (HttpRequestException)
+            {
+                // Retry
+                Task.Delay(TimeSpan.FromMilliseconds(50));
+            }
+        }
+
+        // Unreachable.
+        throw null;
+    }
+}
+```
+
+The preceding code defines a basic retry handler which retries up to five times if a `HttpRequestException` is caught.
+
+During registration, one or more handlers can be added to the configuration for a `HttpClient` via extension methods on the `HttpClientBuilder`.
+
+```csharp
+public static void Configure(IServiceCollection services)
+{
+    services.AddTransient<RetryHandler>();
+
+    services.AddHttpClient("example", c =>
+    {
+        c.BaseAddress = new Uri("https://localhost:5000/");
+    })
+    .AddHttpMessageHandler<RetryHandler>(); // Retry requests to GitHub using the retry handler
+}
+```
+
+In the preceding code, the `RetryHandler` is registered as a transient service with DI. Once registered, `AddHttpMessageHandler` can be called, passing in the type for the handler.
+
+Multiple handlers can be registered in the order that they should execute. Each handler wraps the next handler until the final `HttpClientHandler` executes the request.
+
+```csharp
+public static void Configure(IServiceCollection services)
+{
+    services.AddTransient<OuterHandler>();
+    services.AddTransient<InnerHandler>();
+
+    services.AddHttpClient("example", c =>
+    {
+        c.BaseAddress = new Uri("https://localhost:5000/");
+    })
+    .AddHttpMessageHandler<OuterHandler>() // This handler is on the outside and executes first on the way out and last on the way in.
+    .AddHttpMessageHandler<InnerHandler>(); // This handler is on the inside, closest to the request.
+}
+```
+
+## Handle errors with Polly
+
+_NOTE: This is a WIP since the Polly Extensions API is still currently under design._
```

Comment:
Is this a note to yourself, or would you like the reader to see this? If it's intended for the reader, please use the following syntax:
```
> [!NOTE]
> This section is a work in progress, since the Polly Extensions API is in the design phase.
```